backend/validation_results_model.py

```diff
@@ -1,14 +1,40 @@
-from pynamodb.attributes import BooleanAttribute, UnicodeAttribute
+from enum import Enum
+
+from pynamodb.attributes import UnicodeAttribute
+from pynamodb.indexes import AllProjection, GlobalSecondaryIndex
 from pynamodb.models import Model
 
 from .resources import ResourceName
 
 
+class ValidationResult(Enum):
+    FAILED = "Failed"
+    PASSED = "Passed"
+
+
+# TODO: Remove inherit-non-class when https://github.com/PyCQA/pylint/issues/3950 is fixed
+class ValidationOutcomeIdx(
+    GlobalSecondaryIndex["ValidationResultsModel"]
+):  # pylint:disable=too-few-public-methods,inherit-non-class
+    class Meta:  # pylint:disable=too-few-public-methods
+
+        index_name = "validation_outcome"
+        read_capacity_units = 1
+        write_capacity_units = 1
+        projection = AllProjection()
+
+    pk = UnicodeAttribute(hash_key=True)
+    status = UnicodeAttribute(range_key=True)
```

Comment:
These are basically copies of the actual index defined in infrastructure/processing_stack.py. As discussed we should probably tie them together if possible, so that we're not reimplementing this.